cmd/object-handlers.go

```diff
@@ -600,8 +600,13 @@ func (api objectAPIHandlers) headObjectHandler(ctx context.Context, objectAPI Ob
 		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
 		return
 	}
-
-	if s3Error := checkRequestAuthType(ctx, r, policy.GetObjectAction, bucket, object); s3Error != ErrNone {
+	objAPI := api.ObjectAPI()
+	tags, err := objAPI.GetObjectTags(ctx, bucket, object, opts)
+	if err != nil {
+		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
```

Comment:
This changes the order of errors and likely changes the errors returned when multiple problems exist.

In this case it seems that we "leak" whether the object exists, before we have check auth. This needs careful implementation.

See also `// As per "Permission" section in....` below.